benchmark/poseidon2/poseidon2_config.cc

```diff
@@ -0,0 +1,86 @@
+#include "benchmark/poseidon2/poseidon2_config.h"
+
+#include "tachyon/base/console/iostream.h"
+#include "tachyon/base/flag/flag_parser.h"
+
+namespace tachyon {
+namespace base {
+
+template <>
+class FlagValueTraits<Poseidon2Config::PrimeField> {
+ public:
+  static bool ParseValue(std::string_view input,
+                         Poseidon2Config::PrimeField* value,
+                         std::string* reason) {
+    if (input == "bn254_fr") {
+      *value = Poseidon2Config::PrimeField::kBn254Fr;
+    } else {
+      *reason = absl::Substitute("Unknown test set: $0", input);
+      return false;
+    }
+    return true;
+  }
+};
+
+template <>
+class FlagValueTraits<Poseidon2Config::Vendor> {
+ public:
+  static bool ParseValue(std::string_view input, Poseidon2Config::Vendor* value,
+                         std::string* reason) {
+    if (input == "horizen") {
```

Comment:
why don't we do a switch statement for this too?

Reply:
c++ switch doesn't support string comparision.